Refactor init

yenda opened the pull request:
Refactoring init sequence to remove event fireworks.
Incidently login is a bit faster

# note to reviewers

important: node is restarted everytime user logs in

- [x] refactor intialize account 
- [x] refactor initialize account tests
tdb.
- [x] move init, web3, notifications, signals events and functions to dedicated namespaces
- [x] refactor login flow in `accounts.login.models`

Simplified flow for login:
- event `:ui/login` is dispatched
- node is initialized with user config or default config
- `node.started` signal is received, applying `:login` fx
- `:callback/login` event is dispatched, account is changed
in datastore, web-data is cleared
- `:init/initialize-account` event is dispatched

- [x] replace event dispatches by function composition

# how to test
need extensive tests, mostly around account login, creation and recovery, change network and enable bootnodes, try transactions and web3 browsing.

status: ready

yenda committed:
refactor :initialize-account into single event

fix tests for :initialize-account refactoring
fix destructuring typo

gnl commented on src/status_im/ui/screens/events.cljs:
```diff
@@ -293,56 +309,71 @@
+                network network-status peers-count peers-summary view-id navigation-stack
+                status-module-initialized? status-node-started? device-UUID
+                push-notifications/initial? semaphores]
+         :or   [network (get app-db :network)]} db
```
This should be `:or {network ...}`

yenda committed:
refactor events and fns to dedicated namespaces

- move init events and fns to dedicated namespace
- move web3 events and fns to dedicated namespaces
- move signal events and fns to dedicated namespace
- move notifications events and fns to dedicated namespace
- cosmetic fixes
- remove print statements
- fix formating
- remove unused imports

yenda committed:
refactor login flow

simplified flow:

- event `:ui/login` is dispatched
- node is initialized with user config or default config
- `node.started` signal is received, applying `:login` fx
- `:callback/login` event is dispatched, account is changed
in datastore, web-data is cleared
- `:init/initialize-account` event is dispatched

replace event dispatches by function composition

fix bug in universal links where url to be processed after login
was never removed

Signed-off-by: Eric Dvorsak <eric@dvorsak.fr>

goranjovic reviewed (approved):
lgtm. just remember to squash the commits

lukaszfryc commented:
@yenda there is no send button in chats. Also, the command button does not work when tapped. Other than that it works fine. I did not notice any improvements in login speed.

yenda commented:
@goranjovic I think I will squash it into 3, I would like to keep some granularity in case some bug slips in and we need to bisect, I built and tested at every commit so they are all good

yenda commented:
@lukaszfryc I had to do a complicated rebase with the password saving feature that was merged
android: https://i.diawi.com/MysGZ2
iOS: https://i.diawi.com/ANCkok

<blockquote><div><strong><a href="https://i.diawi.com/MysGZ2">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>
<blockquote><div><strong><a href="https://i.diawi.com/ANCkok">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>
